synapse/models/syn.py

```diff
@@ -197,3 +198,35 @@ def _revModl201710191144(self):
                     i = i + len(chunk)
                     logger.debug('Loading {:,d} [{}%] rows into transaction'.format(i, int((i / tot) * 100)))
         logger.debug('Finished adding node:created rows to the Cortex')
+
+    @modelrev('syn', 201711012123)
+    def _revModl201711012123(self):
+        with self.core.getCoreXact():
+            now = s_common.now()
+            forms = set()
+            adds = []
+            # Lifting all tufo:form rows first and then grabbing the unique
+            # forms referenced allows us to migrate forms which are not loaded
+            # at the time this migration is performed (important in the event
+            # a custom model is loaded) after the Cortex is started.
+            logger.debug('Lifting tufo:form rows')
+            for i, p, v, t in self.core.store.getRowsByProp('tufo:form'):
+                forms.add(v)
+            logger.debug('Computing all node:ndef values')
+            for form in forms:
+                for i, p, v, t in self.core.store.getRowsByProp(form):
+                    # This is quicker than going through the norm process
+                    nv = s_common.guid((p, v))
+                    adds.append((i, 'node:ndef', nv, now))
```

Comment:
may need logic here to smash the adds in at some reasonable increment. maybe every 1 mil or something?